ChronoVisor/src/ChronicleMetaDirectory.cpp

```diff
@@ -40,65 +37,98 @@ bool ChronicleMetaDirectory::create_chronicle(const std::string& name) {
     t2 = std::chrono::steady_clock::now();
     std::chrono::duration<double, std::nano> duration = (t2 - t1);
     LOGD("time in %s: %lf ns", __FUNCTION__, duration.count());
-    return res.second;
+    if (res.second) {
+        return CL_SUCCESS;
+    } else {
+        return CL_ERR_UNKNOWN;
+    }
 }
 
-bool ChronicleMetaDirectory::create_chronicle(const std::string& name,
-                                              const std::unordered_map<std::string, std::string>& attrs) {
+int ChronicleMetaDirectory::create_chronicle(const std::string& name,
+                                             const std::unordered_map<std::string, std::string>& attrs) {
     LOGD("creating Chronicle name=%s", name.c_str());
     Chronicle *pChronicle = new Chronicle();
     pChronicle->setName(name);
     pChronicle->setProperty(attrs);
     uint64_t cid = CityHash64(name.c_str(), name.size());
     pChronicle->setCid(cid);
     auto res = chronicleMap_->emplace(cid, pChronicle);
-    if (chronicleMap_->size() % 10 == 0)
-        LOGD("10 chronicles have been created");
-    return res.second;
+    if (res.second) {
+        return CL_SUCCESS;
+    } else {
```

Comment:
This seems okay, if multiple clients try to create the same chronicle, only one entry is added to the map. But will have a memory leak - delete pChronicle. We should probably add an issue to run valgrind and check for memory leaks in chronolog later.